src/cli/templates/middleware-template.ts

```diff
@@ -13,7 +13,7 @@ export function getMiddlewareTemplate(middlewareName: string): string {
         } from 'medusa-extender';
         import { NextFunction, Response } from 'express';
         
-        @Middleware({ requireAuth: true, routes: [{ method: '', path: '' }] })
+        @Middleware({ requireAuth: true, routes: [{ method: 'all', path: '*' }] })
```

Comment:
Suggestion: when applying a middleware on *, this middleware will act as a custom domain one. It means that requiredAuth will apply the authenticate middleware and the middleware will be loaded before any /admin or /store routes/middleware.
Might be worth adding a comment about that above the middleware during the generation? /admin and /store are applied differently. The three of them have their own loader as well.

Reply:
Wdyt @Saboor-Hakimi ?

Reply:
there is an issue with the indent, is that me or you can see it as well?

Reply:
I have no idea what you're talking about, you're using this thing "dedent" for indentation, I will take a look into it that how it works.

btw there are some indentation issues on the freshly generated code using medex.

Here is an example in the service file which I have generated.
![image](https://user-images.githubusercontent.com/44523544/214606989-ac65aaa5-0704-4294-8f9e-fa71abd07fda.png)

Reply:
![image](https://user-images.githubusercontent.com/44523544/214607102-6a2a7cdd-bb46-451d-afdf-2e3313ee481a.png)

Reply:
The indent is weird indeed, the dedent packages is suppose to remove the indentation link to code indentation that are added to the string.

About the comment I would suggest suggest something like that
```
// Note: When applying this middleware to the wildcard (*) route, it will act as a custom domain middleware.
// This means that if the `requiredAuth` options is applied the admin auth middleware is being choose to be applied. Custom middlewares will be loaded before any routes or middleware in the /admin or /store directories.
// To specifically act on the admin or store domain you can use `/admin/*` or `/store/*`
```

Reply:
@Saboor-Hakimi after the comment above I ll be able to merge

Reply:
I removed the indentations from comments in the exact same way which you asked for.